src/main/java/com/fasterxml/jackson/databind/deser/BasicDeserializerFactory.java

```diff
@@ -520,14 +521,25 @@ private void _addSelectedPropertiesBasedCreator(DeserializationContext ctxt,
                 if (unwrapper != null) {
                     _reportUnwrappedCreatorProperty(ctxt, beanDesc, param);
                 }
-                // Must be injectable or have name; without either won't work
-                if ((name == null) && (injectId == null)) {
-                    ctxt.reportBadTypeDefinition(beanDesc,
+                // [databind#562] Allow @JsonAnySetter in creators
+                if (Boolean.TRUE.equals(ctxt.getAnnotationIntrospector().hasAnySetter(param))) {
+                    if (anySetterIndex >= 0) {
+                        ctxt.reportBadTypeDefinition(beanDesc,
+                                "More than one 'any-setter' specified (parameters #%d vs #%d)",
+                                anySetterIndex, i);
+                    }
+                    anySetterIndex = i;
+                } else {
+                    // Must be injectable or have name; without either won't work
+                    if ((name == null) && (injectId == null)) {
+                        ctxt.reportBadTypeDefinition(beanDesc,
 "Argument #%d of Creator %s has no property name (and is not Injectable): can not use as property-based Creator",
 i, candidate);
+                    }
                 }
             }
-            properties[i] = constructCreatorProperty(ctxt, beanDesc, name, i, param, injectId);
+            properties[i] = constructCreatorProperty(ctxt, beanDesc, name, i, param, injectId,
+                    anySetterIndex >= 0);
```

Comment:
Is this correct? Shouldn't last argument be `anySetterIndex == i`? (otherwise all remaining properties are marked as "any-setters")